### ISSUE TITLE
Indexing out-of-bounds error

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#1016]
With TOT:

```
TEST(NVFuserTest, TMP) {
  Fusion fusion;
  FusionGuard fg(&fusion);
  auto tv0 = makeSymbolicTensor(2);
  fusion.addInput(tv0);
  auto tv1 = add(tv0, new Double(1));
  auto tv2 = add(tv1, new Double(2));
  fusion.addOutput(tv2);
  tv1->setMemoryType(MemoryType::Shared);
  tv2->split(-1, 8);
  fusion.printMath();
  fusion.printKernel();
  FusionExecutor fe;
  fe.compileFusion(&fusion);
  int numel_x = 10;
  int numel_y = 11;
  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
  at::Tensor t0 = at::randn({numel_x, numel_y}, options);
  std::vector<IValue> inputs = {t0};
  auto outputs = fe.runFusion(inputs);
}
```

```
__global__ void CUDAGeneratedKernel(Tensor<float, 2> T0, Tensor<float, 2> T2) {
  alignas(4) extern __shared__ char array[];
  unsigned offset = 0;
  DEFINE_MAGIC_ZERO
  offset = alignBufferSize(offset,4);
  float* T1 = reinterpret_cast<float*>(array + offset);
  offset += ((T0.size[0] * T0.size[1]) * sizeof(float));
  #pragma unroll 1
  for(nvfuser_index_t ki30 = 0; ki30 < T0.size[0]; ++ki30) {
    #pragma unroll 1
    for(nvfuser_index_t ki32 = 0; ki32 < T0.size[1]; ++ki32) {
      T1[(ki30 * T0.size[1]) + ki32]
        = T0[(ki30 * T0.stride[0]) + (ki32 * T0.stride[1])]
        + (float) 1;
    }
  }
  __barrier_sync(0);
  #pragma unroll 1
  for(nvfuser_index_t ki21 = 0; ki21 < T0.size[0]; ++ki21) {
    #pragma unroll 1
    for(nvfuser_index_t ki23 = 0; ki23 < (ceilDiv(T0.size[1], 8)); ++ki23) {
      #pragma unroll
      for(nvfuser_index_t ki25 = 0; ki25 < 8; ++ki25) {
        if ((((ki23 * 8) + (ki25 + nvfuser_zero)) < T0.size[1])) {
          T2[(ki21 * (1 * T0.size[1])) + (((ki23 * 8) + (ki25 + nvfuser_zero)) * 1)]
            = T1[(ki21 * ((ceilDiv(T0.size[1], 8)) * 8)) + ((ki23 * 8) + ki25)]
            + (float) 2;
        }
      }
      UPDATE_MAGIC_ZERO
    }
  }
}
```

The indexing to `T1` as a producer at the second loop nest can move past the allocated boundary due to `((ceilDiv(T0.size[1], 8)) * 8))`, which should be `T0.size[1]`. compute-sanitizer catches the error.

**Comment by csarofeen** [comment]
```.cpp
__global__ void CUDAGeneratedKernel(Tensor<float, 2> T0, Tensor<float, 2> T2) {
  alignas(4) extern __shared__ char array[];
  unsigned offset = 0;
  NVFUSER_DEFINE_MAGIC_ZERO
  offset = alignBufferSize(offset,4);
  float* T1 = reinterpret_cast<float*>(array + offset);
  offset += ((T0.size[0] * T0.size[1]) * sizeof(float));
  #pragma unroll 1
  for(nvfuser_index_t ki68 = 0; ki68 < T0.size[0]; ++ki68) {
    #pragma unroll 1
    for(nvfuser_index_t ki70 = 0; ki70 < T0.size[1]; ++ki70) {
      T1[(ki68 * T0.size[1]) + ki70]
        = T0[(ki68 * T0.stride[0]) + (ki70 * T0.stride[1])]
        + (float) 1;
    }
  }
  __barrier_sync(0);
  #pragma unroll 1
  for(nvfuser_index_t ki59 = 0; ki59 < T0.size[0]; ++ki59) {
    #pragma unroll
    for(nvfuser_index_t ki61 = 0; ki61 < 8; ++ki61) {
      #pragma unroll 1
      for(nvfuser_index_t ki63 = 0; ki63 < (ceilDiv(T0.size[1], 8)); ++ki63) {
        if (((((ki61 + nvfuser_zero) * (ceilDiv(T0.size[1], 8))) + ki63) < T0.size[1])) {
          T2[(ki59 * (1 * T0.size[1])) + ((((ki61 + nvfuser_zero) * (ceilDiv(T0.size[1], 8))) + ki63) * 1)]
            = T1[(ki59 * T0.size[1]) + ((ki61 * (ceilDiv(T0.size[1], 8))) + ki63)]
            + (float) 2;
        }
      }
    }
    NVFUSER_UPDATE_MAGIC_ZERO
  }
}
```

**Comment by naoyam** [comment]
Fixed in #1270
